src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -172,4 +178,41 @@ private void validateColumnForPartition(@Nullable String columnName, @Nullable S
                                                     columnName, type));
     }
   }
+
+  private void validateClusteringOrder(@Nullable Schema schema) {
+    if (!shouldCreatePartitionedTable() || clusteringOrder == null || clusteringOrder.isEmpty() || schema == null) {
+      return;
+    }
+    List<String> columnsNames = Arrays.stream(clusteringOrder.split(",")).map(String::trim)
+      .collect(Collectors.toList());
+    if (columnsNames.size() > MAX_NUMBER_OF_COLUMNS) {
+      throw new InvalidConfigPropertyException(
+        String.format("%s clustering fields specified, exceeding the limit of %s.", columnsNames.size(),
```

Comment:
Why are we using %s instead of %d. Is it working that way?

Expected no more than '%d' clustering fields, found '%d'.

Reply:
Done.